app/src/main/java/org/simple/clinic/drugs/selectionv2/PrescribedDrugWithDosagesSheet.kt

```diff
@@ -23,6 +37,44 @@ class PrescribedDrugWithDosagesSheet : BottomSheetActivity() {
 
     recyclerView.adapter = adapter
     recyclerView.layoutManager = LinearLayoutManager(this)
+    displayDrugName()
+
+    Observable.mergeArray(sheetCreates())
+        .observeOn(io())
+        .compose(controller)
+        .observeOn(AndroidSchedulers.mainThread())
+        .takeUntil(onDestroys)
+        .subscribe { uiChange -> uiChange(this) }
+  }
+
+  override fun onDestroy() {
+    onDestroys.onNext(ScreenDestroyed)
+    super.onDestroy()
+  }
+
+  private fun sheetCreates(): Observable<UiEvent> {
+    val drugName = intent.getStringExtra(KEY_DRUG_NAME)
+    return Observable.just(PrescribedDrugsWithDosagesSheetCreated(drugName))
+  }
+
+  private fun displayDrugName() {
+    val drugName = intent.getStringExtra(KEY_DRUG_NAME)
+    if (drugName.isNullOrEmpty().not()) {
```

Comment:
You can use `drugName.isNotBlank()` instead here. :)

Reply:
Wait, why is this condition even present? Can the name ever be empty? In that case, will we hide the name TextView altogether? 

If you find out that the name can be empty then let's move the logic of hiding the title or using a different title to the controller.

Reply:
You are right. `drugName` cannot be empty.